src/pybind/mgr/volumes/fs/operations/versions/metadata_manager.py

```diff
@@ -42,7 +46,14 @@ def refresh(self):
         conf_data = StringIO()
         try:
             log.debug("opening config {0}".format(self.config_path))
```

Comment:
Dropping a parallel note, on snapshot incarnation retention, based on the conversation in this [comment chain](https://github.com/ceph/ceph/pull/35647/files#r442168572), as shared with @vshankar and @kotreshhr 

**Problem:** When moving subvolume metadata information stored as files within the subvolume, and hence automatically within each of its snapshots, to a key-value store, we cannot rely on the subvolume snapshot carrying a metadata copy as a reference to its incarnation.

This leads to the following requirements.

**Requirements:**
a. Given a (subvolname, snapname) tuple, its incarnation information should be available
- NOTE: It is assumed, that (volume, subvolumegroup) is part of the tuple, but not mentioned explicitly
- For v1 subvolumes, this comes from the subvolume path key
- For v2 subvolumes, this information needs to be stored explicitly, as the current subvolume incarnation/path need not be the same as when the snapshot was created

b. All state data updates, regarding the subvolume-snapshots, should be operated on from a single source
- This information, at present, would include,
  - Snapshots protected
  - Snapshots being cloned

**Current solution for (a):**
- Subvolume snapshot metadata file as a wormhole to its incarnation
  - Currently implemented by reading the incarnation information from the metadata file directly within the subvolume snapshot
  - Alternately, using symlink information of the metadata file within the subvolume snapshot, to read from an incarnation specific metadata file at the subvolume
- Both approaches rely on the snapshot storing a metadata file for reference to its incarnation, and hence, when moving to a key-value store, makes the scheme non-future proof

**Alternate solution(s) for (a):**
- Implicitly infer the snapshot incarnation by inspecting the snapshot directory listing to determine the path
- Explicitly store per snapshot its incarnation information in the meta file and later in the key-value store, in a separate "snapshots" section

**For requirement (b):**
- All state data is already stored and operated on, from the primary metadata file of the subvolume
  - When a subvolume is deleted, but its snapshots retained, this primary metadata files "global' section is only updated as needed, and other sections are retained as is
  - The metadata file within the snapshot is referred to only for the incarnation information (which is being dealt with as part of (a) above)
- This hence could move seamlessly to the key-value store in the future

Reply:
> **Alternate solution(s) for (a):**
> 
>     * Implicitly infer the snapshot incarnation by inspecting the snapshot directory listing to determine the path

This alternative is implemented.

> **For requirement (b):**
> 
>     * All state data is already stored and operated on, from the primary metadata file of the subvolume
>       
>       * When a subvolume is deleted, but its snapshots retained, this primary metadata files "global' section is only updated as needed, and other sections are retained as is
>       * The metadata file within the snapshot is referred to only for the incarnation information (which is being dealt with as part of (a) above)
> 
>     * This hence could move seamlessly to the key-value store in the future

Single meta file is retained with prior metadata.